### ISSUE TITLE
How can I restore croppedAreaPixels and rotation from localStorage

### ISSUE THREAD

**Issue by rokkuminchan** [ricardo-ch/react-easy-crop#282]
### Discussed in https://github.com/ricardo-ch/react-easy-crop/discussions/281

<div type='discussions-op-text'>

<sup>Originally posted by **rokkuminchan** August 20, 2021</sup>
I have created a sample to save CroppedAreaPixels and rotation to locaStorage and restore its data when refresh the page.
Everything works fine when I rotation and zoom the image with the angle of either 0 or 180. But things start to act weirdly when I rotation and zoom with the angle of either 90 or 270. Everytime i refresh the page, the image's zoom is not consistent and the zoom is not maintained. My code is below:
https://codesandbox.io/s/react-easy-crop-example-with-local-storage-sync-forked-z7z2z

App.js
```js
import React, { useState, useEffect } from 'react'
import Slider from '@material-ui/core/Slider'
import Cropper from 'react-easy-crop'
import './App.css'

const App = () => {
  const [crop, setCrop] = useState({ x: 0, y: 0 });
  const [initialCroppedAreaPixels, setInitialCroppedAreaPixels] = useState(undefined);
  const [rotation, setRotation] = useState(0);
  const [zoom, setZoom] = useState(1);

  useEffect(() => {
    const croppedData = JSON.parse(localStorage.getItem('croppedData'));

    if (!croppedData) return;

    setRotation(croppedData.rotation);
    setTimeout(() => {
      setInitialCroppedAreaPixels(croppedData.croppedAreaPixels);
    }, 1);
  }, []);

  const onCropComplete = (croppedArea, croppedAreaPixels) => {
    localStorage.setItem('croppedData', JSON.stringify({ zoom, rotation, croppedAreaPixels }));
  };

  function onRotate() {
    if (rotation === 360) return setRotation(0)

    const value = rotation + 90
    setRotation(value)
  }

  return (
    <div className="App">
      <div className="crop-container">
        <Cropper
          image="https://img.huffingtonpost.com/asset/5ab4d4ac2000007d06eb2c56.jpeg?cache=sih0jwle4e&ops=1910_1000"
          crop={crop}
          zoom={zoom}
          rotation={rotation}
          aspect={4 / 3}
          onCropChange={setCrop}
          onCropComplete={onCropComplete}
          onRotationChange={setRotation}
          onZoomChange={setZoom}
          initialCroppedAreaPixels={initialCroppedAreaPixels}
        />
      </div>
      <div className="controls">
        <Slider
          value={zoom}
          min={1}
          max={3}
          step={0.1}
          aria-labelledby="Zoom"
          onChange={(e, zoom) => setZoom(zoom)}
          classes={{ container: 'slider' }}
        />
        <br />
        <input type="button" onClick={onRotate} value="Rotate" />
      </div>
    </div>
  )
}

export default App;
```

App.css
```css
body {
  margin: 0;
  padding: 0;
}

.App {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.crop-container {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 80px;
}

.controls {
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 50%;
  transform: translateX(-50%);
  height: 80px;
  display: flex;
  align-items: center;
}

.slider {
  padding: 22px 0px;
}
```

Before:
![Screen Shot 2021-08-20 at 10 43 54](https://user-images.githubusercontent.com/71059093/130166067-d5b7f415-a1fc-4bbe-a010-1c9991cd672b.png)

After:
![Screen Shot 2021-08-20 at 10 44 19](https://user-images.githubusercontent.com/71059093/130166079-5e5953ff-7e35-4941-a032-159cdabf9d6d.png)

Did I miss something?</div>

**Comment by ValentinH** [comment]
I don't know if you change it after posting this issue, but your sandbox example works fine when I test it.

**Comment by rokkuminchan** [comment]
> I don't know if you change it after posting this issue, but your sandbox example works fine when I test it.

No, if you rotate the image with the angle of 90 and zoom it to a specific level. After that refresh the page. The zoom level will be different each time

**Comment by ValentinH** [comment]
Indeed, something looks wrong.

To be transparent, I don't have much bandwidth to investigate this issue at the moment. If this is a blocker for you,  I suggest to have a look yourself and try to submit a PR for a fix. 🙂

**Comment by ValentinH** [comment]
#312 might be fixing this as well.

**Comment by ValentinH** [comment]
This is fixed with v4.0.0. Please have a look to the release notes.
